battlepass/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+//      _______  ________  ________  ________   ______   _______   _______
+//    ╱╱       ╲╱        ╲╱        ╲╱        ╲_╱      ╲╲╱       ╲╲╱       ╲╲
+//   ╱╱      __╱         ╱         ╱         ╱        ╱╱        ╱╱        ╱╱
+//  ╱       ╱ ╱         ╱         ╱        _╱         ╱         ╱         ╱
+//  ╲________╱╲___╱____╱╲__╱__╱__╱╲________╱╲________╱╲___╱____╱╲________╱
+//
+// This file is part of GameDAO Protocol.
+// Copyright (C) 2018-2022 GameDAO AG.
+// SPDX-License-Identifier: Apache-2.0
+
+//! BATTLEPASS
+//! This pallet provides functionality to create, manage and participate in battlepasses.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+use frame_support::pallet_prelude::*;
+use frame_system::pallet_prelude::*;
+use sp_std::convert::TryInto;
+use sp_runtime::traits::Hash;
+
+pub mod types;
+pub use types::*;
+
+type String<T> = BoundedVec<u8, <T as pallet::Config>::StringLimit>;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+  
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>>
+			+ IsType<<Self as frame_system::Config>::Event>
+			+ Into<<Self as frame_system::Config>::Event>;
+
+		/// The maximum length of a name or cid stored on-chain.
+		#[pallet::constant]
+		type StringLimit: Get<u32>;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// New BattlePass created
+		BattlepassCreated {
+			// org_id: T::Hash,
+			org_id: T::Hash,
+			battlepass_id: T::Hash,
+			block_number: T::BlockNumber
+		},
+		
+		/// BattlePass claimed
+		BattlepassClaimed {
+			claimer: T::AccountId,
+			org_id: T::Hash,
+			battlepass_id: T::Hash,
+			nft_id: T::Hash,
+			block_number: T::BlockNumber
```

Comment:
The `block_number` is not required, at least for indexing. Squid has an information about the block.

Reply:
removed.